builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -630,46 +720,84 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 	}
 
 	if vL, ok := d.GetOk("disk"); ok {
-		disks := make([]hardDisk, len(vL.([]interface{})))
-		for i, v := range vL.([]interface{}) {
-			disk := v.(map[string]interface{})
-			if i == 0 {
+		if diskSet, ok := vL.(*schema.Set); ok {
+
+			disks := []hardDisk{}
+			hasBootableDisk := false
+			for _, value := range diskSet.List() {
+				disk := value.(map[string]interface{})
+				newDisk := hardDisk{}
+
 				if v, ok := disk["template"].(string); ok && v != "" {
```

Comment:
Similar to my other comment, this is grabbed from the schema which will zero-value out each attribute, so we would get an empty string here instead of a nil.